jwst/extract_1d/extract.py

```diff
@@ -1874,34 +1917,10 @@ def extract(self, data, wl_array, verbose):
             x_array.fill((self.xstart + self.xstop) / 2.)
 
         if self.wcs is not None:
-            nelem = slice1 - slice0
-            if self.exp_type in WFSS_EXPTYPES:
-                # We expect two (x and y) or three (x, y, spectral order).
-                n_inputs = self.wcs.forward_transform.n_inputs
-                ra = np.zeros(nelem, dtype=np.float64)
-                dec = np.zeros(nelem, dtype=np.float64)
-                # Temporary variable so as not to clobber `wavelength`.
-                wcs_wl = np.zeros(nelem, dtype=np.float64)
-                transform = self.wcs.forward_transform
-                if n_inputs == 2:
-                    for i in range(nelem):
-                        ra[i], dec[i], wcs_wl[i], _ = transform(
-                                        x_array[i], y_array[i])
-                elif n_inputs == 3:
-                    for i in range(nelem):
-                        ra[i], dec[i], wcs_wl[i], _ = transform(
-                                x_array[i], y_array[i], self.spectral_order)
-                else:
-                    if verbose:
-                        log.warning("n_inputs for wcs function is %d",
-                                    n_inputs)
-                        log.warning("WCS function was expected to take "
-                                    "either 2 or 3 arguments.")
-                    ra[:] = -999.
-                    dec[:] = -999.
-                    wcs_wl[:] = -999.
-            else:
-                ra, dec, wcs_wl = self.wcs(x_array, y_array)
+            stuff = self.wcs(x_array, y_array)
+            ra = stuff[0]
+            dec = stuff[1]
+            wcs_wl = stuff[2]
```

Comment:
I wondered why you prefer to unpack in separate assignments instead of 
`ra, dec, wcs_wl = self.wcs(x_array, y_array)`

Reply:
`mask` is defined twice in this section, on line 1926 and further on line 1939. 
Also, if this is to cover the NIrspec case, I think it would be best for future development to check for `NaN` in the wavelength arrays, not in the celestial coordinates. 
For example, Nirspec now populates the three arrays (ra, dec, lam) with `NaNs`, but one can argue that this is incorrect because celestial coordinates are always valid. The NaNs are forced only in the wavelength array because some slits fall outside the detector or pixels on the detector fall outside the slit. At some point in the future I plan to change the Nirspec implementation (when I figure out how) and have NaNs only in the wavelength arrays. This will allow us to compute the footprint on the sky for Nirspec slits. I hope this makes sense.

Reply:
I'll change the name of `get_celestial_coordinates`; thanks for pointing that out.
I assigned the output of `wcs` to a variable because the function can return either three or four items, and it's cleaner (I think) to just assign the output to a single variable (a tuple) and then pick out what I need, than to first find out how many values the function returns, have an `if` block to call the function and get either three or four values, and then only use three of them.
I'll change the assignment to `mask` to use wavelength, rather than getting `mask` once for right ascension and again for declination.  I do think it would be better to only have to check one thing (wavelength).